File: packages/c#/Model/Action.cs
using Newtonsoft.Json;

namespace Com.CodeGame.CodeBall2018.DevKit.CSharpCgdk.Model
{
    public sealed class Action
    {
        public double target_velocity_x { get; set; }
        public double target_velocity_y { get; set; }
        public double target_velocity_z { get; set; }
        public double jump_speed { get; set; }
        public bool use_nitro { get; set; }

        [JsonIgnore]
        public Vector target_velocity
        {
            get { return new Vector(target_velocity_x, target_velocity_y, target_velocity_z); }
            set
            {
                target_velocity_x = value.x;
                target_velocity_y = value.y;
                target_velocity_z = value.z;
            }
        }
    }
}

File: packages/rust/src/model/action.rs
#[derive(Debug, Serialize, Deserialize, Copy, Clone)]
pub struct Action {
    pub target_velocity_x: f64,
    pub target_velocity_y: f64,
    pub target_velocity_z: f64,
    pub jump_speed: f64,
    pub use_nitro: bool,
}

impl Default for Action {
    fn default() -> Self {
        Self {
            target_velocity_x: 0.0,
            target_velocity_y: 0.0,
            target_velocity_z: 0.0,
            jump_speed: 0.0,
            use_nitro: false,
        }
    }
}
